python/lancedb/table.py

```diff
@@ -183,13 +194,24 @@ def search(self, query: VEC) -> LanceQueryBuilder:
         return LanceQueryBuilder(self, query)
 
     @classmethod
-    def create(cls, db, name, data, schema=None, mode="create"):
+    def create(cls, db, name, data=None, schema=None, mode="create"):
         tbl = LanceTable(db, name)
-        data = _sanitize_data(data, schema)
-        lance.write_dataset(data, tbl._dataset_uri, mode=mode)
+        if data is not None:
+            data = _sanitize_data(data, schema)
+            lance.write_dataset(data, tbl._dataset_uri, mode=mode)
+        else:
+            # If we're not writing data, we have to manually check that the table doesn't exist
```

Comment:
not sure what logic is here. So it allows user to create table / database without data?
What if the mode == `append` or `overwrite`?

Reply:
Overwrite is only for the case that days exists, so not having data isn't an issue. Append just adds data after any potentially existing data. There's no conflict here.